packages/typespec-vscode/package.json

```diff
@@ -108,6 +108,104 @@
             ],
             "default": "off",
             "description": "Define whether/how the TypeSpec language server should send traces to client. For the traces to show properly in vscode Output, make sure 'Log Level' is also set to 'Trace' so that they won't be filtered at client side, which can be set through 'Developer: Set Log Level...' command."
+          },
+          "typespec.generateCode.emitters": {
+            "scope": "window",
+            "type": "array",
+            "items": {
+              "type": "object",
+              "properties": {
+                "language": {
+                  "type": "string",
+                  "enum": [
+                    ".NET",
+                    "Java",
+                    "JavaScript",
+                    "Python",
+                    "Go",
+                    "OpenAPI3",
+                    "ProtoBuf",
+                    "JsonSchema"
+                  ],
+                  "description": "Define the language the emitter will emit."
+                },
+                "package": {
+                  "type": "string",
+                  "description": "Define the emitter package.\n\nExample (with version): @typespec/http-client-csharp@1.0.0\n\nExample (without version): @typespec/http-client-csharp"
+                },
+                "sourceRepo": {
+                  "type": "string",
+                  "description": "Define the source repository of the emitter package."
+                },
+                "requisites": {
+                  "type": "array",
+                  "items": {
+                    "type": "string"
+                  },
+                  "description": "Define the requisites of the emitter package."
+                },
+                "kind": {
+                  "type": "string",
+                  "enum": [
+                    "client",
+                    "server",
+                    "openapi"
+                  ],
+                  "description": "Define the emitter kind."
+                }
+              }
+            },
+            "default": [
```

Comment:
@RodgeFu, looks like we are baking a list of known 1P emitters in the extension packages. We have discussed this being built-in template driven. I would also like to re-review design of the Generate Code flow. I have comments on what's listed in the design doc on this.

Reply:
See comment: https://microsoft.sharepoint.com/:w:/t/AzureDeveloperExperience/EXfInUWxaOpNpXVtcQOAt4cBoVwvu3tZe3Od__r9vdc8qQ?e=neZ8Dg&nav=eyJjIjoxNzI3NTE5Njc5fQ

Reply:
Sorry that I had a misunderstanding before that I always thought you are talking about scaffolding scenario for template, didn't realize it also extends to the CodeGeneration scenario. But anyway, we are using the hard-coded list here because we have nowhere to read the default list now. And after we have the built-in template available in compiler, I agree that it's a better way to read from there instead of a hard-coded one. I have replied you comment in the doc and we can have more discussion there.

Reply:
please check my comment in the doc. thx.